GVFS/GVFS/CommandLine/ConfigVerb.cs

```diff
@@ -10,18 +11,34 @@ public class ConfigVerb : GVFSVerb.ForNoEnlistment
         private const string ConfigVerbName = "config";
         private LocalGVFSConfig localConfig;
 
+        [Option(
+            'l',
+            "list",
+            SetName = "needNoKeys",

```

Comment:
I don't think this is doing what we need. This `--list` param is part of the set named `needsNoKey`, which makes sense... and yet it still allows you to specify a key and a value. So what is this param set actually achieving?

Likewise the `--delete` flag is in a set named `needsKey`, which is actually incorrect. It's a string property, so it'll get its own value stored in that property, and so it does not need to use the `Key` and `Value` options. And it seems that we can't add a `SetName` to `Option` attributes, so we can't even fix the problem.

Unless I'm missing something here, it doesn't seem like the `SetName` is giving us the validation we want, because both of the following worked, and they should not have:
```
gvfs config --list foo
gvfs config --delete foo bar
```

If we do our own validation, we can quite easily confirm that `--list` and `--delete` are both mutually exclusive of all other `Values` and `Options`.

The only supported usage of this command should be as follows. Anything outside of that should give an error.
```
gvfs config <key>               : get the value for the given key
gvfs config <key> <value>       : set a value for the given key
gvfs config [-l|--list]         : list all keys and their values
gvfs config -d|--delete <key>   : delete the key with the given value
```

Reply:
And we may not want `-l|--list` to be optional either. `git config` on its own produces help text, and you have to be explicit about wanting to list out the settings.

Reply:
That is clear-cut Thank you Saeed. Couple follow-up questions.
1. ```gvfs config <key> <value1> <value2> or more``` should we treat this as error or ignore the additional values.
2. For errors that the Parser doesn't treat as errors (specifying mutually exclusive options, or one like the example above) we would show our own usage text, right.

Reply:
For 1, ideally yes that would be an error, but the CommandLineParser API does not make that easy to detect and it's fairly benign. I think most of our verbs currently allow you to specify ignored extra params, so I would treat that as out of scope for this PR.

For 2, take a look at PrefetchVerb for the pattern. I wouldn't try to replicate the usage text, but rather just show an error.

Reply:
Updated. Screenshot - 

<img width="617" alt="screen shot 2018-10-18 at 11 05 29 am" src="https://user-images.githubusercontent.com/378580/47164368-eceac800-d2c5-11e8-9f51-7b9973cb4e44.png">